### PR TITLE
Address detector list feedback and add feature required state

### PR DESCRIPTION
**Pull request by ohltyler**
*Issue #, if available:*

*Description of changes:*

This PR fixes some small feedback issues for the detector list page and adds a `Feature required` detector state. Specifically:
- Fixes the spacing issue between the detector state filter and index filter when the page is extended horizontally
- Changes the timestamps to capitalize AM/PM
- Shrinks some of the column widths in the table to prevent truncated column titles as much as possible
- Adds a `Feature required` detector state - this is to let the user know that a detector is stopped and cannot run until features are added to it

Before: 

Screen Shot 2020-04-27 at 4 56 50 PM

After: 

Screen Shot 2020-04-27 at 4 56 06 PM

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: public/pages/utils/constants.ts [hunk 1]
```diff
@@ -20,6 +20,7 @@ export enum DETECTOR_STATE_COLOR {
   DISABLED = 'subdued',
   INIT = '#0000cc',
```
Comment (yizheliu-amazon) [inline public/pages/utils/constants.ts:21]: can you try 'primary' for INIT? I recalled 'primary' is blue color as well.

File: server/routes/ad.ts [hunk 2]
```diff
@@ -461,6 +460,16 @@ const getDetectors = async (
...
+      /*
+        If a detector has no features, set to a feature required state
+      */
+      if (
+        detectorState.state === DETECTOR_STATE.DISABLED &&
+        finalDetectors[i].featureAttributes.length === 0
+      ) {
+        detectorState.state = DETECTOR_STATE.FEATURE_REQUIRED;
+      }
     });
```
Comment (yizheliu-amazon) [inline server/routes/ad.ts:473]: you may encapsulate such logic to a method with appropriate name like `getStateFromRawResponse()` or better name

Comment (ohltyler) [reply server/routes/ad.ts:473]: Sure, good idea. Will add a function in adHelpers.

File: public/pages/utils/constants.ts (hunk 1 above)
Comment (ohltyler) [reply public/pages/utils/constants.ts:21]: I see, yes. Will change.

**Pull request merged**